crates/core/src/dht/actions/hold_aspect.rs

```diff
@@ -1,28 +1,56 @@
 use crate::{
     action::{Action, ActionWrapper},
     context::Context,
+    dht::aspect_map::AspectMap,
     instance::dispatch_action,
 };
 use futures::{future::Future, task::Poll};
 use holochain_core_types::{error::HolochainError, network::entry_aspect::EntryAspect};
+use holochain_persistence_api::cas::content::AddressableContent;
+use lib3h_protocol::data_types::EntryListData;
 use snowflake::ProcessUniqueId;
 use std::{pin::Pin, sync::Arc};
 
 pub async fn hold_aspect(aspect: EntryAspect, context: Arc<Context>) -> Result<(), HolochainError> {
-    let action_wrapper = ActionWrapper::new(Action::HoldAspect(aspect.clone()));
-    dispatch_action(context.action_channel(), action_wrapper.clone());
     let id = ProcessUniqueId::new();
-    HoldAspectFuture {
-        context,
-        aspect,
+    let action_wrapper = ActionWrapper::new(Action::HoldAspect((aspect.clone(), id)));
+    dispatch_action(context.action_channel(), action_wrapper.clone());
+    let r = HoldAspectFuture {
+        context: context.clone(),
+        //        aspect,
         id,
     }
-    .await
+    .await;
+    if r.is_err() {
+        error!("HoldAspect action completed with error: {:?}", r);
+    } else {
+        // send a gossip list with this aspect in it back to sim2h so it know we are holding it
+        let c = context.clone();
+        let closure = async move || {
+            let state = context
+                .state()
+                .expect("No state present when trying to respond with gossip list");
+
+            let mut address_map = AspectMap::new();
+            address_map.add(&aspect);
```

Comment:
Huh, this sends back a GossipList with just this one item in it? Isn't this causing sim2h to be confused about the gossip list of this node?

I really think we shouldn't do this (managing results of holding workflows) here at all and instead have sim2h properly (if it isn't) poll for gossip lists.

Reply:
Currently the gossiplist is additive on the sim2h side so this really acts as an acknowledgement of the hold_aspect request.

Reply:
Sim2h's state knowledge is append-only.

We were previously setting "agent-holding" when we sent the hold request, but that could fail - it's much more accurate for core to send a sparse holding list that results in an additional hold record.